src/desktop_notifier/backends/base.py

```diff
@@ -143,46 +143,43 @@ async def get_capabilities(self) -> frozenset[Capability]:
         """
         ...
 
-    def handle_clicked(
-        self, identifier: str, notification: Notification | None = None
-    ) -> None:
-        if notification and notification.on_clicked:
+    def handle_clicked(self, identifier: str) -> None:
+        notification: Notification | None = self._clear_notification_from_cache(identifier)
+        if not notification:
+            return
```

Comment:
This actually changes the logic and makes it impossible to handle notification clicks when no cached instance is available, for example in case of:

1. Push notifications triggered by a remote server.
2. Notifications that we interacted with while the app was not running and that triggered app startup.

Being able to handle callbacks purely using an identifier allows apps for example to make requests to server to fetch information based on the ID, or look at a local database of scheduled notifications and look up how to react to a notification from there.